File: app/src/main/java/de/westnordost/osmagent/quests/Quest.java
package de.westnordost.osmagent.quests;

import de.westnordost.osmagent.quests.osm.ElementGeometry;
import de.westnordost.osmapi.map.data.LatLon;

/** Represents one task for the user to complete/correct */
public interface Quest
{
	Long getId();
	void setId(long id);

	LatLon getMarkerLocation();

	ElementGeometry getGeometry();

	QuestType getType();

	QuestStatus getStatus();

	void setStatus(QuestStatus status);
}

File: app/src/main/java/de/westnordost/streetcomplete/data/quest/Quest.kt
package de.westnordost.streetcomplete.data.quest

import de.westnordost.streetcomplete.data.osm.geometry.ElementGeometry
import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
import de.westnordost.streetcomplete.screens.main.map.components.Pin

/** Represents one task for the user to complete/correct  */
abstract class Quest {
    abstract val key: QuestKey
    abstract val position: LatLon
    abstract val markerLocations: Collection<LatLon>
    abstract val geometry: ElementGeometry

    abstract val type: QuestType

    /** caching pins in the quest allows for faster setting of pins */
    var pins: List<Pin>? = null
}
